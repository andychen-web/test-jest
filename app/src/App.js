import React, { useState } from 'react';
import Control from './components/Control';
import Content from './components/Content';
import { Container } from './Style.css';

const App = () => {
    const [list, setList] = useState([]);
    return (
        <Container>
            <Control
                list={list}
                setList={setList}
            />
            <Content
                list={list}
                setList={setList}
            />
        </Container>
    );
}

export default App;
