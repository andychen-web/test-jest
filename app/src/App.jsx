import React, { useState } from "react";
import Control from "./components/Control.js";
import Content from "./components/Content.js";
import { Container, Title } from "./Style.css.js";
import { TEST_ID } from "./config/testId.js";
import TestLodash from "./components/TestLodash.jsx";
import TestFormPage from "./views/TestFormPage.jsx";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import TestAlgorithm from "./components/TestAlgorithm.jsx";

// test only
const App = () => {
  return (
    <Router>
      <Container>
        {/* <Title data-testid={TEST_ID.APP_TITLE}>Test List Sample</Title>
    <Control list={list} setList={setList} />
  <Content list={list} setList={setList} /> */}
        {/* create Route for below */}
        <Routes>
          <Route path="/test-form" element={<TestFormPage />} />
          <Route path="/test-lodash" element={<TestLodash />} />
          <Route path="/test-algorithm" element={<TestAlgorithm />} />
        </Routes>
      </Container>
    </Router>
  );
};

export default App;
