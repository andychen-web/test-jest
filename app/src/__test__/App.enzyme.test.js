import { shallow, mount } from 'enzyme';
import App from '../App';
import { TEST_ID } from '../config/testId';

describe('Todo App container tests', () => {
    let app;
    const renderApp = () => <App />;
    beforeEach(() => {
        app = mount(renderApp());
    });

    afterEach(() => {
        app.unmount();
    });

    test('Todo App container should display as expected', () => {
        const appShallow = shallow(renderApp());
        expect(appShallow).toMatchSnapshot();
        expect(app).toMatchSnapshot();
    });

    test('App title should display as expected', () => {
        const appTitle = app.find(`div[data-testid="${TEST_ID.APP_TITLE}"]`).text();
        expect(appTitle).toBe('Todo List Sample');
    });

    test('Todo item should add on todo list as expected', () => {
        const todoInput = app.find('input');
        const addButton = app.find('button');
        todoInput.simulate('change', { target: { value: 'todo1' } });
        addButton.simulate('click');

        const todoItem1Content = app.find('li').text();
        expect(todoItem1Content).toBe('todo1');
    });
});
