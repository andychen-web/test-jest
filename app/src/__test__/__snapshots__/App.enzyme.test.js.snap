// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo App container tests Todo App container should display as expected 1`] = `
<styled.div>
  <styled.div
    data-testid="APP_TITLE"
  >
    Todo List Sample
  </styled.div>
  <Control
    list={Array []}
    setList={[Function]}
  />
  <Content
    list={Array []}
    setList={[Function]}
  />
</styled.div>
`;

exports[`Todo App container tests Todo App container should display as expected 2`] = `
<App>
  <styled.div>
    <div
      className="sc-bdvvaa crVrEO"
    >
      <styled.div
        data-testid="APP_TITLE"
      >
        <div
          className="sc-hKwCoD jeypMi"
          data-testid="APP_TITLE"
        >
          Todo List Sample
        </div>
      </styled.div>
      <Control
        list={Array []}
        setList={[Function]}
      >
        <styled.div>
          <div
            className="sc-gsDJrp wlSJp"
          >
            <styled.input
              data-testid="TODO_INPUT"
              onChange={[Function]}
              value=""
            >
              <input
                className="sc-dkPtyc gtRBDb"
                data-testid="TODO_INPUT"
                onChange={[Function]}
                value=""
              />
            </styled.input>
            <styled.button
              data-testid="ADD_BUTTON"
              onClick={[Function]}
            >
              <button
                className="sc-eCImvq jEtYTQ"
                data-testid="ADD_BUTTON"
                onClick={[Function]}
              >
                Add
              </button>
            </styled.button>
          </div>
        </styled.div>
      </Control>
      <Content
        list={Array []}
        setList={[Function]}
      >
        <styled.div>
          <div
            className="sc-gsDJrp wlSJp"
          >
            <styled.ul>
              <ul
                className="sc-jRQAMF kKsXMP"
              />
            </styled.ul>
          </div>
        </styled.div>
      </Content>
    </div>
  </styled.div>
</App>
`;
